doc/asab/web/authn.rst

```diff
@@ -0,0 +1,75 @@
+Authn module
+=================
+
+.. py:currentmodule:: asab.web.authn
+
+Module :py:mod:`asab.web.authn` provides middlewares and classes to allow only authorized users access specified web server endpoints.
+It also allows to forward requests to the authorization server via instance of OAuthForwarder.
+
+Currently available authorization technologies include OAuth 2.0, public/private key and HTTP basic auth.
+
+Middleware
+------------
+
+.. py:method:: authn_middleware_factory(app, implementation, *args, **kwargs):
+
+First step is to define the authorization implementation, which can be the OAuth 2.0, public/private key or
+HTTP basic auth. Depending on the implementation, there are arguments which further specify the authorization
+servers that are going to be used.
+
+When it comes to OAuth 2.0, there are :any:`methods` for every OAuth 2.0 server, that is going to be used for authorization
+and obtainment of the user identity. The relevant method is selected based on access token prefix, that is received
+from the client in the HTTP request (:any:`Authorization` header):
+
+`Authorization: Bearer <OAUTH-SERVER-ID>-<ACCESS_TOKEN>`
+
+The following example illustrates how to register a middleware inside the web server container with OAuth 2.0
+implementation and GitHub method.
+
+.. code:: python
+
+    container.WebApp.middlewares.append(
+        asab.web.authn.authn_middleware_factory(self,
+            "oauth2client",
```

Comment:
@PremyslCerny Could this be with comment like
```
            "oauth2client",  # "pubkeyauth" or "basicauth"
```

Reply:
@SukiCZ Sure, good point. :)